cmd/bucket-lifecycle.go

```diff
@@ -155,9 +155,14 @@ type newerNoncurrentTask struct {
 	versions []ObjectToDelete
 }
 
+type transitionTask struct {
```

Comment:
Outside of `internal/bucket/lifecycle` package, can we use `tier` in place of `storageclass` where possible. It helps us keep terms used in code consistent with the docs?